applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_implicit_dynamic_solver.py

```diff
@@ -184,19 +184,19 @@ def _create_linear_solver(self):
                     if (linear_solver_name == "AMGCL" or linear_solver_name == "AMGCLSolver"):
                         amgcl_param = KM.Parameters("""
                         {
-                            "solver_type" : "AMGCL",
-                            "smoother_type":"ilu0",
-                            "krylov_type": "lgmres",
-                            "coarsening_type": "aggregation",
-                            "max_iteration": 100,
-                            "provide_coordinates": false,
-                            "gmres_krylov_space_dimension": 100,
-                            "verbosity" : 1,
-                            "tolerance": 1e-6,
-                            "scaling": false,
-                            "block_size": 3,
+                            "solver_type"                    : "AMGCL",
+                            "smoother_type"                  : "gauss_seidel",
+                            "krylov_type"                    : "fgmres",
+                            "coarsening_type"                : "smoothed_aggregation",
+                            "max_iteration"                  : 1000,
+                            "provide_coordinates"            : false,
+                            "gmres_krylov_space_dimension"   : 1000,
+                            "verbosity"                      : 1,
+                            "tolerance"                      : 1e-6,
+                            "scaling"                        : false,
+                            "block_size"                     : 3,
                             "use_block_matrices_if_possible" : true,
-                            "coarse_enough" : 500
+                            "coarse_enough"                  : 1000
```

Comment:
on line 203 you are NOT using the factory mechanism for constructing the linear solver, and the big problem is that by doing it this way you do not allow the user to change the defaults for the amgcl solver (or to use a different linear solver)

Reply:
The factory is used in the base solver, this is the contact solver, which derives from mechanical solver, the factory is used there. Here I am just changing the solver default parameters (instead of using the ones set by the factory)

Reply:
https://github.com/KratosMultiphysics/Kratos/blob/c0fed84296a989ea943f78b6ef52de7ef9842f3c/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py#L425 Here, and my solvers depends on this solver